src/lib.rs

```diff
@@ -344,11 +371,27 @@ pub struct Evaluator<'tcx> {
 
     /// The random number generator to use if Miri
     /// is running in non-deterministic mode
-    pub(crate) rng: Option<StdRng>
+    pub(crate) rng: Option<StdRng>,
+
+    /// Extra information needed for unwinding
+    /// We create this even in abort mode, so
+    /// that we can perform some basic validation
+    /// during panics
+    pub(crate) cached_data: CachedTypes<'tcx>,
+
+    /// Whether or not we are currently unwinding from
+    /// a panic
+    pub(crate) unwinding: bool,
+    pub(crate) box_me_up_tmp_ptr: Option<MPlaceTy<'tcx, Tag>>
```

Comment:
Ah, what a glorious hack. ;)  But I see that you will need something like this, to perform some action after `box_me_up` returned.

But isn't this the same as the return place of the box_me_up frame? So, you should be able to just have a special `bool` in that frame's state (which you already do!), and then in `pop_stack` read that boolean to determine that this is the right frame, and then extract the place from that frame's return place. So no global state like this should be needed.